fix: namespace support

imiric opened the pull request:
Hi!

This is the work @asandvig proposed in #110 but didn't submit a PR for it.

I think the feature is very useful, so here it is, but @asandvig should be considered the author.

I'm not familiar with Expat, so let me know if there's a better approach.

I also included changes to the tests, so everything should be passing.

Cheers,

Ivan

martinblech commented:
Thanks for this contribution, it looks promising! There's a couple of things that need to be fixed before merging:
1. Tests are failing for Jython in the [Travis build](https://travis-ci.org/martinblech/xmltodict/builds/134104382). I can't merge this if it breaks the build.
2. I don't see any test cases for round-trip support. Doesn't this require any changes in `unparse()`?

imiric commented:
OK, I fixed the broken Jython build. It was due to inconsistencies in the namespace prefix values passed to the `StartNamespaceDeclHandler` in the different Expat implementations, so I settled on defining a blank namespace as `''` instead of `None` for all versions.

I also added namespace handling to `unparse()`/`_emit()`, and a test for that too.

I'm not proud of this code, but let me know if it matches your expectations.

Cheers,

Ivan

martinblech commented:
Thank you!
